### PR TITLE
replace entrypoints use with importlib.metadata

### PR DESCRIPTION
**Pull request by abn**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -50,7 +50,6 @@ cachecontrol = { version = "^0.12.9", extras = ["filecache"] }
 cachy = "^0.3.0"
 cleo = "^1.0.0a5"
 crashtest = "^0.3.0"
-entrypoints = "^0.4"
```
Comment (Secrus) [inline pyproject.toml:53]: if you are removing this, remove `types-entrypoints` from dev section too

File: src/poetry/console/commands/self/show/plugins.py [hunk 2]
```diff
@@ -8,15 +8,32 @@
...
-    plugins: list[EntryPoint] = dataclasses.field(default_factory=list)
-    application_plugins: list[EntryPoint] = dataclasses.field(default_factory=list)
+    plugins: list[metadata.EntryPoint] = dataclasses.field(default_factory=list)
+    application_plugins: list[metadata.EntryPoint] = dataclasses.field(
+        default_factory=list
+    )
+
+    def append(self, entry_point: metadata.EntryPoint) -> None:
+        from poetry.plugins.application_plugin import ApplicationPlugin
+        from poetry.plugins.plugin import Plugin
```
Comment (Secrus) [inline src/poetry/console/commands/self/show/plugins.py:26]: any particular reason for these imports to be here and not at the top of the file?

File: src/poetry/plugins/plugin_manager.py [hunk 3]
```diff
@@ -41,11 +39,16 @@ def load_plugins(self, env: Env | None = None) -> None:
...
-        return entry_points
+    ) -> list[metadata.EntryPoint]:
+        return [
+            ep
+            for ep in metadata.entry_points(group=self._group)
+            if env is None
+            or (
+                ep.dist is not None
+                and env.site_packages.find_distribution(ep.dist.name) is not None
+            )
```
Comment (Secrus) [inline src/poetry/plugins/plugin_manager.py:50]: I would rewrite this to a normal loop. in its current form, this is barely readable.

**Review by Secrus — CHANGES_REQUESTED** [review]

File: src/poetry/console/commands/self/show/plugins.py (hunk 2 above)
Comment (abn) [reply src/poetry/console/commands/self/show/plugins.py:26]: it is mostly done to defer the import so the load times are quicker; for this change I am simply maintaining the status quo. Any changes to import ordering should be dealt with outside of this PR i reckon.

File: pyproject.toml (hunk 1 above)
Comment (abn) [reply pyproject.toml:53]: Good catch.

File: src/poetry/plugins/plugin_manager.py (hunk 3 above)
Comment (abn) [reply src/poetry/plugins/plugin_manager.py]: Split the conditional into another method for now.

**Review by Secrus — APPROVED** [review]

**Pull request merged**